apps/homescreen/js/datastore.js

```diff
@@ -198,7 +201,9 @@
           (event) => {
             var cursor = event.target.result;
             if (cursor) {
-              results.push(cursor.value);
+              if (filter(cursor.value)) {
```

Comment:
Is this for searching pinned pages? Shouldn't we create an index and search into the datastore using it instead of getting all and filtering afterwards?